client/src/components/JurisdictionAdmin/generateSheets.test.tsx

```diff
@@ -276,4 +309,238 @@ describe('generateSheets', () => {
+      const batches: IBatch[] = [
+        {
+          id: 'B1',
+          name: allStarLyrics,
+          numBallots: 0,
+          auditBoard: {
+            id: 'AB1',
+            name: allStarLyrics,
+          },
+          resultTallySheets: [],
+        },
+        {
+          id: 'B2',
+          name: allStarLyrics,
+          numBallots: 0,
+          auditBoard: {
+            id: 'AB1',
+            name: allStarLyrics,
+          },
+          resultTallySheets: [],
+        },
+      ]
+      const contestChoices: ICandidate[] = [
+        { id: 'C1', name: allStarLyrics, numVotes: 0 },
+        { id: 'C2', name: allStarLyrics, numVotes: 0 },
+        { id: 'C3', name: allStarLyrics, numVotes: 0 },
+        { id: 'C4', name: allStarLyrics, numVotes: 0 },
+        { id: 'C5', name: allStarLyrics, numVotes: 0 },
+        { id: 'C6', name: allStarLyrics, numVotes: 0 },
+      ]
+      const jurisdictionName = allStarLyrics
+      const pdf = await downloadBatchTallySheets(
+        batches,
+        contestChoices,
+        jurisdictionName
+      )
+      await expect(Buffer.from(pdf)).toMatchPdfSnapshot({
+        tolerance: diffTolerance,
+      })
+      expect(mockSavePDF).toHaveBeenCalledWith('Batch Tally Sheets.pdf', {
+        returnPromise: true,
+      })
+    })
+
+    it('Handles long content with no spaces', async () => {
+      const manyAs =
+        'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
+      const batches: IBatch[] = [
+        {
+          id: 'B1',
+          name: manyAs,
+          numBallots: 0,
+          auditBoard: {
+            id: 'AB1',
+            name: manyAs,
+          },
+          resultTallySheets: [],
+        },
+        {
+          id: 'B2',
+          name: manyAs,
+          numBallots: 0,
+          auditBoard: {
+            id: 'AB1',
+            name: manyAs,
+          },
+          resultTallySheets: [],
+        },
+      ]
+      const contestChoices: ICandidate[] = [
+        { id: 'C1', name: manyAs, numVotes: 0 },
+        { id: 'C2', name: manyAs, numVotes: 0 },
+      ]
+      const jurisdictionName = manyAs
+      const pdf = await downloadBatchTallySheets(
+        batches,
+        contestChoices,
+        jurisdictionName
+      )
+      await expect(Buffer.from(pdf)).toMatchPdfSnapshot({
+        tolerance: diffTolerance,
+      })
+      expect(mockSavePDF).toHaveBeenCalledWith('Batch Tally Sheets.pdf', {
+        returnPromise: true,
+      })
+    })
+
+    it('Adds page break after table if necessary - 1', async () => {
+      const pdf = await downloadBatchTallySheets(
+        mockBatches,
+        constructContestChoices(6),
+        mockJurisdiction.name
+      )
+      await expect(Buffer.from(pdf)).toMatchPdfSnapshot({
+        tolerance: diffTolerance,
+      })
+    })
+
+    it('Adds page break after table if necessary - 2', async () => {
```

Comment:
I've covered every page break point reasonably possible through this sequence of `Adds page break after table if necessary - #` tests

Reply:
Realized I could consolidate them in a loop here: https://github.com/votingworks/arlo/pull/1482/commits/d5b1a0bd8af498770053ab50ef82f6096f385c28